sympy/utilities/decorator.py

```diff
@@ -193,37 +195,13 @@ def public(obj):
     return obj
 
 
-class ClassPropertyDescriptor(object):
-    """
-    Helper class for ``classproperty`` decorator.
-    """
-    def __init__(self, fget):
-        self.fget = fget
-
-    def __get__(self, obj, klass=None):
-        if klass is None:
-            klass = type(obj)
-
-        return self.fget.__get__(obj, klass)()
-
-
-def classproperty(func):
-    """
-    This decorator can be used to define a property of a class
-    which is intended to be accessed using class and objects both.
-
-    Examples
-    ========
-
-    >>> from sympy.utilities.decorator import classproperty
-    >>> class Q(object):
-    ...     @classproperty
-    ...     def a_property(self):
-    ...         return 2
-    >>> Q.a_property
-    2
-    """
-    if not isinstance(func, (classmethod, staticmethod)):
-        func = classmethod(func)
-
-    return ClassPropertyDescriptor(func)
+def memoize_property(storage):
+    """Create a property, where the lookup is stored in ``storage``"""
```

Comment:
This docstring is missing the following information:
- That it is a decorator.
- Where the property's name is taken from.
- Where the property's value is taken from.
- Whether the decorated function is allowed to have any parameters.
- Caveats, if applicable.
From the code, I have a vague idea what this does, and if it does what it thinks it's something that I've been needing for quite a while, but I'm not sure that I didn't misinterpret anything. Also, since the docstring is so meager, it's unclear what the function is supposed to do, so if there are any bugs around this function, people will never know whether the bug is in its usages or in its implementation.